tests/unit/FileHelperTest.php

```diff
@@ -0,0 +1,325 @@
+     *
+     * @return void
+     */
+    public function testDeleteDirWithFiles() {
+        // Set up a virual folder structure
+        $structure = [
+            'folder_1' => [
+                'file_1.txt' => 'file_1.txt',
+                'file_2.txt' => 'file_2.txt',
+            ],
+            'folder_2' => [
+                'file_3.txt' => 'file_3.txt',
+                'file_4.txt' => 'file_4.txt',
+                'folder_3' => [
+                    'file_5.txt' => 'file_5.txt',
+                    'file_6.txt' => 'file_6.txt',
+                    
+                ],
+            ],
+        ];
+        $vfs = vfsStream::setup('root');
+        vfsStream::create($structure, $vfs);
+        $filepath = vfsStream::url('root/folder_1');
+
+        // Check vfsStream set up the directories correctly
+        $expected = true;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // folder_1 should now be gone
+        FilesHelper::delete_dir_with_files($filepath);
+        $expected = false;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // Delete a nested folder
+        $filepath = vfsStream::url('root/folder_2/folder_3');
+        FilesHelper::delete_dir_with_files($filepath);
+        // And confirm the top level one still exists
+        $filepath = vfsStream::url('root/folder_2');
+        $expected = true;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+    }
+
+    /**
+     * Test getListOfLocalFilesByDir method
+     *
+     * @return void
+     */
+    public function testGetListOfLocalFilesByDir() {
+        // Set up a virual folder structure
+        $structure = [
+            'folder_1' => [
+                'file_1.jpg' => 'file_1.jpg',
+                'file_2.jpg' => 'file_2.jpg',
+            ],
+            'folder_2' => [
+                'file_3.jpg' => 'file_3.jpg',
+                'file_4.jpg' => 'file_4.jpg',
+                'folder_3' => [
+                    'file_5.jpg' => 'file_5.jpg',
+                    'file_6.jpg' => 'file_6.jpg',
+                    
+                ],
+            ],
+        ];
+        $vfs = vfsStream::setup('root');
+        vfsStream::create($structure, $vfs);
+        // Set virtual WP root directory to /root/ for this test
+        $mock = Mockery::mock('overload:\WP2Static\SiteInfo');
+        $mock->shouldreceive('getPath')->andReturn(vfsStream::url('root') . "/");
+
+
+        // Top level folder
+        $filepath = vfsStream::url('root/folder_1');
+        $expected = ['/folder_1/file_1.jpg', '/folder_1/file_2.jpg'];
+        $actual = FilesHelper::getListOfLocalFilesByDir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // Nested folder
+        $filepath = vfsStream::url('root/folder_2/folder_3');
+        $expected = ['/folder_2/folder_3/file_5.jpg', '/folder_2/folder_3/file_6.jpg'];
```

Comment:
Can we add some unicode filenames, long filenames, etc here to push boundaries?

Reply:
Added through https://github.com/WP2Static/wp2static/pull/638/commits/75ebb9a5a4592957bf4354289eb3a8a5adcb563e. While adding these I found some issues:

* getListOfLocalFilesByDir() doesn't return a complete list of files in a directory because it calls filePathLooksCrawlable() which has file extension and filename restrictions. Unsure if this is intentional or not. Recommend adding a warning in that methods docblock. I wrote a comment about this in one of the tests.
* getListOfLocalFilesByDir() does a urlencode() on filepath but not filename. I have a feeling this isn't the behavior that we want. Recommend fixing the function to urlencode() filename as well and adding information to the docblock describing that this will happen.

Let me know your thoughts.

Reply:
so more like `getCrawlableFiles()`? 

rather than doc blocks, I'd prefer if we add a line in the code, assigning to aptly named variable, ie `$encoded_filename=blah`. The tests act as the real source of truth as to what's happening. The code will give a good idea, too. If the functions are too long that such assigned var gets lost from mental space, we can break them down further. I just don't want to rely on a docblock, which adds lines but does little to protect/enforce behaviour. 

I'd start with a failing test for the encoded filename part and confirm what happens currently and confirm if that failure represents in a real world test of unicode/special chars filepath in WP